### PR TITLE
ci: convert circle jobs to github actions

### PR DESCRIPTION
**Pull request by scurker**
This change will migrate us away from using circle and using github actions instead. There was already a mix of circle/github and this just keeps everything in one spot.

It is currently expected that Circle will be failing on this PR and that the required checks will not run. Once this PR is merged, we will update the branches to require the new github actions checks and remove any integrations with circle. All of the github actions that do run will need to pass on this PR before merging.

### CODE REVIEW

File: .github/workflows/create-release-pr.yml [hunk 1]
```diff
@@ -10,9 +10,6 @@ jobs:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - uses: actions/setup-node@v4
-        with:
-          node-version: 18
```
Comment (thuey) [inline .github/workflows/create-release-pr.yml:15]: mostly just curious, but why did you remove this?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          PACKAGE_VERSION="$(npm pkg get version | tr -d \")"
+          npm version "$PACKAGE_VERSION-canary.${GITHUB_SHA:0:8}" --allow-same-version --no-git-tag-version
+          npm publish --tag=next
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_AUTH_TOKEN }}
+      - name: Publish @deque/cauldron-react
+        run: |
+          cd packages/react
+          PACKAGE_VERSION="$(npm pkg get version | tr -d \")"
+          npm version "$PACKAGE_VERSION-canary.${GITHUB_SHA:0:8}" --allow-same-version --no-git-tag-version
```
Comment (thuey) [inline .github/workflows/release.yml:43]: Should this be in the `canary` job instead of the `publish` job?

Comment (scurker) [reply .github/workflows/release.yml:43]: Whoops, good catch! 🤦 

Looks like I swapped them around accidentally.

File: .github/workflows/create-release-pr.yml (hunk 1 above)
Comment (scurker) [reply .github/workflows/create-release-pr.yml]: Got a little too aggressive 🗡️

**Review by thuey — APPROVED** [review]
